### ISSUE TITLE
eslint-plugin-html plugin fails to load

### ISSUE THREAD

**Issue by pdagosto** [AtomLinter/linter-eslint#1109]
## Issue Type

Question

## Issue Description

After upgrading to version 8.4.1 the linter claims that it can't find the html plugin, which is installed as a dev dependency (version 4.0.2). My .eslintrc.json is where it always was (~/.atom/packages/linter-eslint/node_modules/.bin) and does have the "plugins":["html"] item in it.

## Bug Checklist
- [x] Restart Atom
I've done this, plus re-installed the plugin and re-started Atom again

- [x] Verify the `eslint` CLI gives the proper result, while `linter-eslint` does not
Yes, this is the case as far as I can tell (see debug output below)

- [x] Paste the output of the `Linter Eslint: Debug` command from the Command Palette below

```
Linter Eslint: Debug output here
```  eslint:cli Running on files +0ms
  eslint:glob-util Creating list of files to process. +0ms
  eslint:ignored-paths Looking for ignore file in /home/myuserid/repos/nextgen_workflow/bfx-workflow +0ms
  eslint:ignored-paths Loaded ignore file /home/myuserid/repos/nextgen_workflow/bfx-workflow/.eslintignore +0ms
  eslint:ignored-paths Adding /home/myuserid/repos/nextgen_workflow/bfx-workflow/.eslintignore +0ms
  eslint:cli-engine Linting complete in: 34ms +0ms

(note that no error is reported while in the IDE the <template> tags in my *.vue files are flagged as having an "unexpected character" ('<')

**Comment by Arcanemagus** [comment]
> My .eslintrc.json is where it always was (~/.atom/packages/linter-eslint/node_modules/.bin)

This is _entirely_ wrong, I'm not sure how that would _ever_ work for you 😮.

Were you trying to create a "default" configuration? As explained in the [ESLint documentation](https://eslint.org/docs/user-guide/configuring) the correct way to do this is to place the configuration file at `~/` (eg. `/home/myuserid/.eslintrc.json`).

Without the output of the `Linter Eslint: Debug` command from the Command Palette _within Atom_, I can't really say more, but you should absolutely fix the location of your configuration file first.

**Comment by pdagosto** [comment]
Yes, I am trying to create a "default" configuration (actually I only ever want to use one configuration and I fail to see how I would benefit from having to copy it or link to it in every project).

I moved the config file to my home directory as suggested. Same result in the IDE (plugin not found <template> tag flagged for unexpected token).

I don't remember why the config file was under the .atom tree but that did work. I wouldn't know why.

I re-ran eslint from within the command window in Atom (as I did for output provided in my original post, not sure why you're saying it wasn't present) but this time the result is consistent with what's going on in the IDE:

myuserid@localhost input]$ ~/.atom/packages/linter-eslint/node_modules/.bin/eslint --debug myfile.vue
  eslint:cli Running on files +0ms
  eslint:glob-util Creating list of files to process. +0ms
  eslint:ignored-paths Looking for ignore file in /home/myuserid/repos/nextgen_workflow/bfx-workflow/src/components/input +0ms
  eslint:ignored-paths Could not find ignore file in cwd +1ms
  eslint:cli-engine Processing /home/myuserid/repos/nextgen_workflow/bfx-workflow/src/components/input/myfile.vue +0ms
  eslint:cli-engine Linting /home/myuserid/repos/nextgen_workflow/bfx-workflow/src/components/input/myfile.vue +1ms
  eslint:config Constructing config file hierarchy for /home/myuserid/repos/nextgen_workflow/bfx-workflow/src/components/input +0ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading /home/myuserid/repos/nextgen_workflow/bfx-workflow/package.json +2ms
  eslint:config-file Loading package.json config file: /home/myuserid/repos/nextgen_workflow/bfx-workflow/package.json +0ms
  eslint:config-file Loading JSON config file: /home/myuserid/repos/nextgen_workflow/bfx-workflow/package.json +1ms

  eslint:config Using personal config file +30ms
  eslint:config Using personal config +0ms
  eslint:config-file Loading JSON config file: /home/myuserid/.eslintrc.json +28ms
  eslint:plugins Failed to load plugin eslint-plugin-html. +0ms

Oops! Something went wrong! :(

ESLint: 4.17.0.
ESLint couldn't find the plugin "eslint-plugin-html". This can happen for a couple different reasons:

1. If ESLint is installed globally, then make sure eslint-plugin-html is also installed globally. A globally-installed ESLint cannot find a locally-installed plugin.

2. If ESLint is installed locally, then it's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm i eslint-plugin-html@latest --save-dev

If you still can't figure out the problem, please stop by https://gitter.im/eslint/eslint to chat with the team.

Note: I did try re-installing eslint-plugin-html. The output you see is after I did that and re-started Atom.

**Comment by Arcanemagus** [comment]
I'm talking about the Command Palette in Atom, _not_ your terminal. You can access this under Packages -> Command Palette -> Toggle. It should look like this:
![image](https://user-images.githubusercontent.com/427137/36283133-5bf0a726-1258-11e8-92cc-c31b5cca3c30.png)

----

It looks like you are trying to use the fallback `eslint` that is bundled with this package, that is only intended to be a _last_ resort to attempt to get syntax checking at least, but we are planning on removing for _exactly_ the problems you are bringing up here: users attempting to use it as their _only_ ESLint. 😆

The recommended method of installing and using ESLint is to have it as a `devDependency` on your project, that way _everyone_ working on it can get the exact same linter configuration... including a future you coming back to the project after a while.

If you don't want to maintain that, you can install ESLint globally, but note that then _every_ project that doesn't have a local version will be using the same version no matter what. You will also need to install all plugins globally using this method as well.

**Comment by skylize** [comment]
[quote]
I have always relied on the ESLint installed along with the Atom package as specified in the settings for the plugin:

If there is no local installation Atom will use the built-in ESLint in the linter-eslint package itself.
[/quote]

It's very problematic trying to resolve a proper location for your config file with the built-in copy of eslint. As has already been mentioned, this "feature" is already planned for deprecation for this reason. 

We continue to recommend separate installations and configs per project. This will prove quite valuable when you start trying to share projects with other people.

But if you are going to use a single system-wide installation and config, you should globally install ESLint with `npm install --global eslint`, and also globally install any plugins you need, and place the config file in your user home directory.

**Comment by pdagosto** [comment]
Thanks for the replies - much appreciated.

I wasn't aware that I was relying on a soon-to-be -deprecated feature. I understand the concern about sharing projects however that is not relevant in my case. Sharing is not ( and won't be) a concern. Also, the concept of creating a separate configuration for every project runs completely counter to my experience (mostly emacs). I've never created a separate .emacs file or installed separate instances of emacs plugins in every project directory (no, not even in team environments). I guess you have your reasons but to me its not how editor configs and plugins should work.

However, if that's how Atom works I will follow the scheme for using a global installation as you outlined. I would add that the documentation for the plugin settings should be, how can I put this, more illuminating on this point than it is now.

I will give this another try tomorrow.

**Comment by skylize** [comment]
>  I would add that the documentation for the plugin settings should be, how can I put this, more illuminating on this point than it is now.

Thanks for the input. We are happy to look at pull requests. 😉

**Comment by skylize** [comment]
This Issue has been quiet for a while and appears to be mostly resolved. Closing as Resolved, but let us know if you have any further questions or concerns.
